### PR TITLE
Mainnet review fixes

### LINKED ISSUE #8: Virtual functions?
Why are all of these functions marked as virtual?
I can't imagine any reason that they would be imported into another contract, so no need to mark them as virtual.

https://github.com/EBOGDAO/og-dapp/blob/0a2e2ecfae21dc4677f6c4cc6a17992286c2f362/contracts/Agreement.sol#L74-L111

### LINKED ISSUE #9: Use onlyRole instead of only member
https://github.com/EBOGDAO/og-dapp/blob/0a2e2ecfae21dc4677f6c4cc6a17992286c2f362/contracts/Agreement.sol#L43-L46

You should use the onlyRole() modifier from OZ access control, rather than implementing your own onlyMember() modifier.

Use `onlyRole("OG")` anywhere that you have used `onlyMember()`.

### LINKED ISSUE #11: leaveCommunity() should be internal
https://github.com/EBOGDAO/og-dapp/blob/0a2e2ecfae21dc4677f6c4cc6a17992286c2f362/contracts/Agreement.sol#L104-L106

When would a user call this function?
It seems like it should only be called by the `optOutOfDAO()` function, in which case it should be internal.
That being the case, it probably makes more sense to just move `renounceRole("OG", msg.sender);` to the end of the `optOutOfDAO()` function.

### PR DESCRIPTION
**Pull request by stinkyfi**
Resolved issues #8 and #9
Tested on Rinkeby: 0xab5cAA579410Dd0eCfbB7dd11759350BE2d01A93

### CODE REVIEW

File: contracts/Agreement.sol [hunk 1]
```diff
@@ -71,8 +64,7 @@ contract Agreement is AccessControl, Ownable {
       return optedOutAccounts;
     }
 
-    function getAddressStatus(address _address) public view returns (string memory) {
-      require(isMember(msg.sender), "Restricted to members.");
+    function getAddressStatus(address _address) public view onlyRole("OG") returns (string memory) {
```
Comment (auryn-macmillan) [inline contracts/Agreement.sol]: Why is this `onlyRole("OG") ?

I don't see any good reason to restrict it.

**Review by auryn-macmillan — CHANGES_REQUESTED** [review]

File: contracts/Agreement.sol (hunk 1 above)
Comment (stinkyfi) [reply contracts/Agreement.sol]: I was converting all of the `require(isMember(...` to `onlyRole` per issue #9 

Should any address be able to check the OG Membership status? Or should only OG DAO members be able to validate that?

Comment (auryn-macmillan) [reply contracts/Agreement.sol]: I don't see any reason to restrict it. I think you can safely remove this check.

Comment (stinkyfi) [reply contracts/Agreement.sol]: Adjustment made

**Review by auryn-macmillan — APPROVED** [review]
lgtm

**Pull request merged**